src/outputOPAT.ts

```diff
@@ -0,0 +1,92 @@
+import { ValidatorResult } from "./ValidatorResult";
+import * as fs from "fs";
+import Handlebars from "handlebars";
+
+export function outputOPAT(
+  data: any,
+  catalogData: any,
+  outputFile: string
+): ValidatorResult {
+  try {
+    const templateString = fs
+      .readFileSync("templates/opat-0.1.0.handlebars")
+      .toString();
+    const template = Handlebars.compile(templateString);
+
+    if (catalogData) {
+      data.catalog = catalogData;
+    }
+    const date = new Date();
+    data.now = date.toLocaleDateString();
+
+    Handlebars.registerHelper("chapterOrTableTitle", function (chapter) {
+      if (chapter.order < 4) return "Table";
+      else return "Chapter";
+    });
+
+    Handlebars.registerHelper(
+      "catalogCriteriaLabel",
+      function (chapterId, criteriaNum) {
+        for (const chapter of catalogData.chapters) {
+          /* istanbul ignore else */
+          if (chapter.id === chapterId) {
+            for (const catalogChapterCriteria of chapter.criteria) {
+              if (catalogChapterCriteria.id === criteriaNum) {
+                return catalogChapterCriteria.label;
+              }
+            }
+          }
+        }
+      }
+    );
+
+    Handlebars.registerHelper("catalogComponentLabel", function (componentId) {
+      for (const component of catalogData.components) {
+        if (component.id === componentId) {
+          return component.label;
+        }
+      }
+    });
+
+    Handlebars.registerHelper("levelLabel", function (level) {
+      let label = "";
+      switch (level) {
```

Comment:
I had some thoughts about translation/internationalization on #64 - as a consequence of that, I think ultimately perhaps we do want these levels to live in the catalog after all, but we can discuss :)

Reply:
Ya, I realized after doing this templating that the levels should have been in the catalog. Definitely on my radar as well to fix.

Reply:
Created a ticket in the backlog: #70